### ISSUE TITLE
Settings are not saved

### ISSUE THREAD

**Issue by z33ky** [hogsy/OniFoxed#8]
I thought this was a Linux issue or something, but looking closer at the code I think this is true for Windows as well.
If you delete the "persist.dat", start the game and modify settings, they will not be remembered the next time you open Oni.

The `ONrPersist()` function is supposed to store these settings in the "persist.dat".
https://github.com/hogsy/OniFoxed/blob/9362f444058f99f92038a2733cdc741fa09c90f6/OniProj/OniGameSource/Oni_Persistance.c#L152-L163
However, [`BFrFile_FOpen()`](https://github.com/hogsy/OniFoxed/blob/9362f444058f99f92038a2733cdc741fa09c90f6/BungieFrameWork/BFW_Source/BFW_FileManager/BFW_FileManager_Common.c#L855) will error out since it uses [`BFrFileRef_Set()`](https://github.com/hogsy/OniFoxed/blob/9362f444058f99f92038a2733cdc741fa09c90f6/BungieFrameWork/BFW_Source/BFW_FileManager/Platform_Win32/BFW_FileManager_Win32.c#L163) which returns `BFcError_FileNotFound` if the file doesn't already exist.
https://github.com/hogsy/OniFoxed/blob/9362f444058f99f92038a2733cdc741fa09c90f6/BungieFrameWork/BFW_Source/BFW_FileManager/Platform_Win32/BFW_FileManager_Win32.c#L162-L184

Indeed there doesn't seem to be a good way to create a file at all in the provided API.
Looking at how [`BFrFile_Create()` is used](https://github.com/hogsy/OniFoxed/search?q=BFrFile_Create) it seems likely that creating a `BFtFileRef` to a non-existing file shouldn't actually error out.

**Comment by madebr** [comment]
Indeed, it seems to me that `BFrFileRef_Set` should not error when the file does not exist.
See this usage:
https://github.com/hogsy/OniFoxed/blob/9362f444058f99f92038a2733cdc741fa09c90f6/BungieFrameWork/BFW_Source/BFW_FileManager/BFW_FileManager_Common.c#L102-L108

It expects `UUcError_None` on success + does an extra test for existence for that file.
